### DISCUSSION TITLE
How to exclude certain parts of the road network?

### DISCUSSION THREAD

**Discussion by kumbraGrr** [valhalla/valhalla#4520] [Q&A] [answered]
Dear people, is there a way to exclude some road based on the osm tag or location for route calculation?

Best so far I could find is `exclude_locations ` parameter for route endpoint. But this is not ideal for my case. I am ideally looking for something like Lua configuration where I could specify based on the osm tags which roads to exclude when considering roads for route and apply rules without need to rebuild network. Does anyone have recommendation where to look or in which direction? 

I am also aware that I could do edit of osm/pbf files prior building graphs but this could not work on the fly as I would have to rebuild network for every change.

Thank you!

**Comment by nilsnolde** [comment, marked as answer]
Nope, sorry, we don't support that kind of use case (yet?). Some of it would be fairly easy to support (e.g. road class, though we do reclassify some, e.g. ferries, which would certainly lead to unwanted results) but also has the potential to be a performance sink in case you exclude high class roads, i.e. including shortcuts. There's also a PR to exclude bridge & tunnel segments which isn't merged yet.

**Reply by kumbraGrr** [reply]
Thank you. I just thought maybe someone figured out workaround for it or it already exists but I missed it somehow.
Will follow up updates and MRs on this topic. 

Thank you once again for a fast answer

**Reply by nilsnolde** [reply]
always worth asking:)
